live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/transform/request/ATSTransformRequesterTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.request;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+import static org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.request.ATSTransformRequester.TIMEOUT_KEY;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.IntegrationProperties;
+import org.alfresco.hxi_connector.live_ingester.adapters.config.properties.Transform;
+import org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.model.ClientData;
+import org.alfresco.hxi_connector.live_ingester.adapters.messaging.transform.request.model.ATSTransformRequest;
+import org.alfresco.hxi_connector.live_ingester.domain.ports.transform_engine.TransformRequest;
+
+@ExtendWith(MockitoExtension.class)
+public class ATSTransformRequesterTest
+{
+    private static final String NODE_REF = "nodeRef";
+    private static final String TARGET_MIME_TYPE = "targetMimeType";
+    private static final int TIMEOUT = 123;
+    private static final int ATTEMPT = 2;
+    private static final String QUEUE_NAME = "queueName";
+
+    @Mock
+    IntegrationProperties integrationProperties;
+    @InjectMocks
+    ATSTransformRequester atsTransformRequester;
+
+    @Test
+    public void testToTransformRequest_missingConfigurationGivesNoError()
+    {
+        // given
+        setUpIntegrationProperties(null);
```

Comment:
Since transform options are annotated `@NotNull` in `Transform` class, missing (`null`) configuration will likely cause application fail to start. Maybe following setup would be more suitable here (or parametrize that and have both `null` and `emptyMap()` as input?)
```suggestion
        setUpIntegrationProperties(emptyMap());
```

Reply:
I did try using the application with no options provided and it seemed to work:
```
...
alfresco:
  repository:
    endpoint: activemq:topic:alfresco.repo.event2
  bulk-ingester:
    endpoint: activemq:queue:bulk-ingester-events
  transform:
    request:
      endpoint: activemq:queue:acs-repo-transform-request?jmsMessageType=Text
      timeout: 20000
    response:
      queue-name: org.alfresco.hxinsight-connector.transform.response
      ...
```